src/backend/distributed/utils/node_metadata.c

```diff
@@ -620,27 +684,44 @@ AddNodeMetadata(char *nodeName, int32 nodePort, int32 groupId, char *nodeRack,
 		}
 	}
 
+	/* if nodeRole hasn't been added yet there's a constraint for one-node-per-group */
+	if (nodeRole != InvalidOid)
+	{
+		primaryRole = PrimaryNodeRoleId();
+		if (nodeRole == primaryRole)
```

Comment:
That's a good metric in general. I think here it'd be a little overkill, but consider the comment optional.